### PR TITLE
Fix #1893: Correct woi param for non-gregorian calendars.

### LINKED ISSUE #1893: DAU pings for non-gregorian calendar shows wrong `woi`
### Steps to Reproduce 
  1. Set calendar to Japanese in iOS general settings
  2. Launch the app

**Actual result:** 
woi parameter is wrong

**Expected result:**

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

**Brave Version:** 

**Device details:** 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes issue #1893 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

Can't really by tested by QA

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by iccub** [comment]
Blocking until Joel's review

**Review by kylehickinson — APPROVED** [review]

File: BraveSharedTests/DAUTests.swift [hunk 1]
```diff
@@ -140,6 +140,16 @@ class DAUTests: XCTestCase {
...
     }
     
+    func testMondayOfCurrentWeekFormatted() {
+        // Making sure gregorian year is showing
+        let yearString = Date().mondayOfCurrentWeekFormatted?.truncate(length: 4, trailing: "")
+        let year = Int(yearString!)!
+        
+        // There is no easy way to mock a different calendar in unit tests, just checking if
+        // year we get looks 'correct' should be good enough.
+        XCTAssert(year > 2018 && year < 2025)
```
Comment (kylehickinson) [inline BraveSharedTests/DAUTests.swift:150]: TODO: Come back in 2025 and fix this test 😂

**Comment by kylehickinson** [comment]
Didn't realize you were waiting for @jhreis explicitly. I'll defer merging until after

File: BraveSharedTests/DAUTests.swift (hunk 1 above)
Comment (iccub) [reply BraveSharedTests/DAUTests.swift:150]: I hope they will add calendar setting to test plans in the future 🤡atm there's only language and region

**Review by jhreis — APPROVED** [review]

**Pull request merged**